src/main/java/com/rultor/agents/IndexesRequests.java

```diff
@@ -40,7 +49,95 @@
  */
 public final class IndexesRequests implements SuperAgent {
     @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public void execute(final Talks talks) throws IOException {
-        throw new UnsupportedOperationException();
+        int max = this.max(talks);
+        for (final Talk talk : talks.active()) {
+            max = this.update(talk, max);
+        }
+    }
+
+    /**
+     * Sets the index value of a request node, if it exists and if the talk node
+     *  has archive/log nodes.
+     * @param talk Talk, to which the index node should be appended.
+     * @param max Maximal index value among all talks
+     * @return New maximal value. If an index node was added, it is equal to
+     *  (max+1), otherwise - max.
+     * @throws IOException Thrown in case of XML parsing errors.
+     */
+    private int update(final Talk talk, final int max) throws IOException {
```

Comment:
I would implement this method differently. This way:

```
talk.modify(
  new Directives().xpath("/talk/request").attr("index", max)
);
return max + talk.read().nodes("/talk/request/@index").size();
```

That's enough. First, we set the `@index` to the `max`. If XPath doesn't find anything, it won't set anything. Then, we count how many requests were updated. According to `talk.xsd`, there could be only one `request`, so this `size()` will return either 0 or 1. Make sense?